handler.js

```diff
@@ -360,117 +373,26 @@ class GithubBuild {
       return null;
     }
 
     return [
-      // matched "uat" or "functional"
+      // matched "uat", "functional" or "deployment"
       matches[1],
 
       // matched argument
       matches[2]
     ];
   }
 
+  // get all tests related to the given commands
   getTests() {
-    let tests = [];
-    const branch = this.getBranch();
-
-    if(branch === "release" || branch === "master") {
-      tests.push({
-        name: "js-php",
-        type: "unit-tests",
-        deployable: false
-      });
-      // no need to run tests when pushing to release
-      // should be only the case when merging a PR on release
-    } else if (this.enableUatAndFunctionalTests()) {
-      // Only enabled for PR against release
-
-      console.log("deploy and run all tests on release");
-      tests.push({
-        name: "js-php",
-        type: "unit-tests",
-        deployable: false
-      });
-      tests.push({
-        name: "backend",
-        type: "uat",
-        deployable: true
-      });
-
-      tests.push({
-        name: "frontend",
-        type: "uat",
-        deployable: false
-      });
-
-      tests.push({
-        name: "functional",
-        type: "functional",
-        deployable: false
-      });
-    } else {
-      const commitMsg = this.getCommitMsg();
-      const forceCommand = this.getForceCommand();
-      const skipDeployment = commitMsg.indexOf("[skip deployment]") !== -1;
-      const skipUnitTests = commitMsg.indexOf("[skip unit-tests]") !== -1;
-
-      console.log(commitMsg, forceCommand, skipDeployment, skipUnitTests);
-
-      // run unit-tests only if there is no skip command
-      if (!skipUnitTests) {
-        tests.push({
-          name: "js-php",
-          type: "unit-tests",
-          deployable: false
-        });
-      }
-
-      // check if there is a forcecommand and if we have enabled them
-      if(forceCommand && this.enableForceUATCommands()) {
-        const forceType = forceCommand[0];
-        const forceArgument = forceCommand[1];
-
-        if(typeof forceArgument !== "undefined") {
-          tests.push({
-            name: forceArgument.trim(),
-            type: forceType,
-            deployable: !skipDeployment
-          });
-        } else if (forceType === "uat") {
-          tests.push({
-            name: "backend",
-            type: "uat",
-            deployable: !skipDeployment
-          });
-
-          tests.push({
-            name: "frontend",
-            type: "uat",
-            deployable: false
-          });
-
-          tests.push({
-            name: "functional",
-            type: "functional",
-            deployable: false
-          });
-        } else if (forceType === "functional") {
-          tests.push({
-            name: "functional",
-            type: "functional",
-            deployable: !skipDeployment
-          });
-        }
-      }
-    }
-
-    return tests;
+    const ruleSet = new Rule(this.dataSet)
```

Comment:
- There is no class named Rule. 
- Should there be semicolon ";" at the end of the statement?